frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/create-sc.tsx

```diff
@@ -46,20 +47,32 @@ const makeAutoDiscoveryCall = (
 
   fetchK8s(LocalVolumeDiscovery, DISCOVERY_CR_NAME, LOCAL_STORAGE_NAMESPACE)
     .then((discoveryRes: K8sResourceKind) => {
-      const nodes = new Set(
-        discoveryRes?.spec?.nodeSelector?.nodeSelectorTerms?.[0]?.matchExpressions?.[0]?.values,
-      );
-      selectedNodes.forEach((name) => nodes.add(name));
-      const patch = [
-        {
-          op: 'replace',
-          path: `/spec/nodeSelector/nodeSelectorTerms/0/matchExpressions/0/values`,
-          value: Array.from(nodes),
-        },
-      ];
-      return k8sPatch(LocalVolumeDiscovery, discoveryRes, patch);
+      const nodeSelectorTerms = discoveryRes?.spec?.nodeSelector?.nodeSelectorTerms;
+      const [selectorIndex, expIndex] = nodeSelectorTerms
+        ? getLabelIndex(nodeSelectorTerms, HOSTNAME_LABEL_KEY)
+        : [-1, -1];
+      if (selectorIndex !== -1 && expIndex !== -1) {
+        const nodes = new Set(
+          discoveryRes?.spec?.nodeSelector?.nodeSelectorTerms?.[selectorIndex]?.matchExpressions?.[
+            expIndex
+          ]?.values,
+        );
+        selectedNodes.forEach((name) => nodes.add(name));
+        const patch = [
+          {
+            op: 'replace',
+            path: `/spec/nodeSelector/nodeSelectorTerms/${selectorIndex}/matchExpressions/${expIndex}/values`,
+            value: Array.from(nodes),
+          },
+        ];
+        return k8sPatch(LocalVolumeDiscovery, discoveryRes, patch);
+      }
+      throw new Error('Failed to update the Auto Detect Volume!');
```

Comment:
This doesn't look like the part of this PR as per description.

Reply:
can you explain what are we trying to do over here?

Reply:
This is required to handle the CLI scenario i..e when auto discovery is being created from cli and getting updated from UI. 
For that UI need to find out the index of `HOSTNAME_LABEL_KEY` key in order to update that

Reply:
Updated the description of the PR too

Reply:
Is it not required in generic feature?